### PR TITLE
Header Refactor

### PR DESCRIPTION
**Pull request by vuonga1103**
### Changes:
- Added Settings stub and route.
- Fixed issue with drawer not closing on sidebar “Settings” click. 
- Ensured proper display in mobile Landscape mode, in both default and logged in versions.
- Removed JOIN button on mobile view when on log in or register route.
- Refactored Header to utilize useMediaQuery() to set mobile/desktop responsiveness.
- Created FemmecubatorLogo, MobileHeader, DesktopHeader components.
- Remove window resize listeners from App.js
- Fixed error in Header test.

**Updates**
- Fixed memory leak error in the registration component and login component that is caused by updating the context API when the component is in the process of being unmounted. Deleted the Auth Context API. Will leave the Global Provider Context API for possible later use.

- Fixed broken unit test as a result of previous fix

image

Co-authored By: @cfernando415

### CODE REVIEW

File: src/components/Header/Header.test.js [hunk 1]
```diff
@@ -33,13 +33,12 @@ describe('<Header />', () => {
...
-        <GlobalProvider>
-          <Router>
-            <Header />
-          </Router>
-        </GlobalProvider>
-      </AuthProvider>
+      <Router>
+        <Header>
+          <DesktopHeader />
+          <MobileHeader />
```
Comment (cfernando415) [inline src/components/Header/Header.test.js:39]: not needed

**Review by cfernando415 — CHANGES_REQUESTED** [review]
pls see comment

File: src/components/Header/Header.test.js (hunk 1 above)
Comment (vuonga1103) [reply src/components/Header/Header.test.js]: done 687a2c1

**Review by cfernando415 — APPROVED** [review]
👍

File: src/components/Header/Header.js [hunk 2]
```diff
@@ -207,64 +69,20 @@ function Header() {
     }
```
Comment (cfernando415) [inline src/components/Header/Header.js:69]: let's put this in async method within the useEffect

File: src/components/Header/Header.js [hunk 3]
```diff
@@ -285,157 +104,90 @@ function Header() {
...
 
-  const getAccountChoices = () => {
-    if (menuHeaders && menuHeaders.length) {
-      return menuHeaders
-        .filter(
-          ({ href }) => href === '/login?logout=true' || href === '/settings'
-        )
-        .map(({ id, href: to, label }) => {
+  const getAccountChoices = useCallback(() => {
+    const logoutHandler = () => Auth.logoff();
```
Comment (cfernando415) [inline src/components/Header/Header.js:114]: delete, we're going to pass Auth.logoff as a callback in onClick

File: src/components/Header/Header.js [hunk 4]
```diff
@@ -285,157 +104,90 @@ function Header() {
...
+            <MenuItem
               key={id}
               {...{
-                component: RouterLink,
-                to,
-                color: 'inherit',
-                style: { textDecoration: 'none' },
-                onClick: handleAccountClose,
+                className: isMobile ? drawerChoice : accountChoice,
+                onClick: logoutHandler,
```
Comment (cfernando415) [inline src/components/Header/Header.js:124]: Auth.logoff

File: src/components/Header/Header.js (hunk 3 above)
Comment (cfernando415) [reply src/components/Header/Header.js:114]: not sure if this is just a stylistic preference, but I prefer to minimize nested arrow functions unless it's a useEffect hook or a functional arrow component

**Review by cfernando415 — CHANGES_REQUESTED** [review]
pls see additional comment

File: src/components/Header/Header.js (hunk 4 above)
Comment (vuonga1103) [reply src/components/Header/Header.js]: resolved in db8dbe8

File: src/components/Header/Header.js (hunk 3 above)
Comment (vuonga1103) [reply src/components/Header/Header.js]: resolved in db8dbe8

File: src/components/Header/Header.js (hunk 2 above)
Comment (vuonga1103) [reply src/components/Header/Header.js]: resolved in 174adae

**Review by cfernando415 — APPROVED** [review]
👍

**Pull request merged**